File: backend/B2BApplication/Models/Article.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace B2BApplication.Models
{
    public class Article
    {
        [Key]
        public int ArticleID { get; set; }
        public string ArticleCode { get; set; }
        public string ArticleIntitule { get; set; }
        public decimal ArticlePrixHT { get; set; }
        [JsonIgnore]
        public string imagePath { get; set; }
        public string imageContent {  get; set; }
        public string description { get; set; }
        //decimal
        public Taxe? Taxe { get; set; }
      
        public int? TaxeId { get; set; }
        public string ArticleBarCode { get; set;}
        public bool ArticleEtat { get; set; } = true;
        public bool ArticleStatistique { get; set; } = true;
        public Catalogue Catalogue { get; set; }
        public int CatalogueId { get; set; }
        public Famille Famille { get; set; }
        public int FamilleId { get; set; }
        public DateTime DateCreate { get; set; } = DateTime.Now;
        public DateTime DateUpdate { get; set; } = DateTime.Now;
        public User UserCreated { get; set; }
        public User UserModified { get; set; }
        public List<Tarif> tarifs {  get; set; }
        public int nbArticleSold { get; set; }
        public List<ArticleViews> ViewedBy { get; set; }
    }
}

File: frontend/src/app/models/article.ts
import { Catalogue } from "./catalogue"
import { Family } from "./family"
import { Tax } from "./tax"
import { User } from "./user"

export class Article{
    articleID:number
    articleCode:string
    articleIntitule:string
    articlePrixHT:number
    taxe:Tax
    articleBarCode:string
    articleEtat:boolean
    articleStatistique:boolean
    catalogue:Catalogue
    famille:Family
    dateCreate:Date
    dateUpdate:Date
    userCreated:User
    UserModified:User
    nbArticleSold:number
    imagePath:string
    imageContent:string=""
    description=""
    constructor(articleID:number,articleCode:string,articleIntitule:string,articlePrixHT:number,taxe:Tax,articleBarCode:string,
        articleEtat:boolean,articleStatistique:boolean,catalogue:Catalogue,famille:Family,
        dateCreate:Date,dateUpdate:Date,userCreated:User,UserModified:User,imagePath:string){
this.articleID=articleID
this.articleCode=articleCode
this.articleIntitule=articleIntitule
this.articlePrixHT=articlePrixHT
this.taxe=taxe
this.articleBarCode=articleBarCode
this.articleEtat=articleEtat
this.articleStatistique=articleStatistique
this.catalogue=catalogue
this.famille=famille
this.dateCreate=dateCreate
this.dateUpdate=dateUpdate
this.UserModified=UserModified
this.userCreated=userCreated
this.nbArticleSold=0
this.imagePath=imagePath
        }
}